packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -167,9 +168,19 @@ const NativeSelect = ({
   ...customProps
 }) => {
   const [uncontrolledValue, setUncontrolledValue] = useState(!isValuePresent(value) ? defaultValue || getFirstValue(options, isFilterStyle) : undefined);
-  const refIsControlled = useRef(isValuePresent(value));
   const refSelect = useRef();
   const theme = React.useContext(ThemeContext);
+  const [refIsControlled, setrefIsControlled] = useState(useRef(isValuePresent(value)));
+
+  useEffect(() => {
+    // added this to allow initial null values
+    if (value) {
+      setrefIsControlled(true);
+      setUncontrolledValue('');
+    } else {
+      setrefIsControlled(false);
+    }
+  }, [value]);
```

Comment:
I don't think we need state to identify controlled version here !! the issue is due to use of `useRef` to store `isControlled` boolean value, because of useRef `refIsControlled` is initialized only once and ignored for the re-renders that happens with prop updates. 

So i think just removing useRef from line No.170 and keeping normal variable would fix this issue of `not updating value prop on re-renders`.

Reply:
Updated the cod with out usage of state variable. Thanks